corehq/motech/repeaters/models.py

```diff
@@ -965,6 +965,17 @@ def iter_partition(self, start, partition, total_partitions):
     def get_domains_with_records(self):
         return self.order_by().values_list("domain", flat=True).distinct()
 
+    def get_repeat_record_ids(self, domain, repeater_id=None, state=None, payload_id=None):
```

Comment:
> This should live in the RepeatRecordManager and shouldn't have
validation logic in it.

I'm ambivalent about this. In theory this makes sense, but how bad would it be if code that *did* pull all of a domain's repeat records made it past code review & QA and into production? If that could impact the entire system, I'd consider doing the validation in here anyway.

Reply:
That's probably a reasonable consideration, though I don't have any sense of how bad it might actually be. @gherceg do you have thoughts here?

Reply:
Sorry just seeing this. With the domain filter, returning all repeat records for a domain isn't _terrible_ and is a similar query to what is done on the repeat record report page if no filters are specified. Granted that isn't the most performant query at the moment, but I don't think we are at risk of having system wide impacts from these queries.

Reply:
okay, cool